### PR TITLE
893 many to many match

### PR DESCRIPTION
**Pull request by osborj**
## Pre-review checklist

I have ensured that:

- [x] Code runs successfully in the development environment.
- [x] New code is tested to prove that it works. If not, justify reasoning.
- [x] New code is documented using [type hinting](https://realpython.com/lessons/type-hinting/) and following the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) docstring format.
- [x] New code follows the project naming standards.

## Post-review checklist

As I merge these changes, I have:

- [ ] Updated the related Jira ticket, including approximate time spent (including review).
- [ ] Moved the related Jira ticket to Done.

### CODE REVIEW

File: cishouseholds/merge.py [hunk 1]
```diff
@@ -222,3 +222,100 @@ def many_to_one_antibody_flag(df: DataFrame, column_name_to_assign: str, group_b
...
+    df = df.withColumn(
+        failed_flag_column_name_to_assign,
+        F.when(
+            F.last("classification_different_to_first").over(window).isNotNull()
+            & (F.col("identify_many_to_many_flag") == 1),
+            1,
+        ).otherwise(None),
+    )
+
+    # record_processed set to 1 if evaluated and drop flag to be set, 0 if evaluated and drop flag to be None,
```
Comment (osborj) [inline cishouseholds/merge.py:290]: to be set *to one*

File: cishouseholds/merge.py [hunk 2]
```diff
@@ -222,3 +222,100 @@ def many_to_one_antibody_flag(df: DataFrame, column_name_to_assign: str, group_b
...
+    )
+
+    window = Window.partitionBy(group_by_column, "identify_many_to_many_flag")
+
+    if process_type == "antibody":
+        column_to_validate = "antibody_test_result_classification"
+    elif process_type == "swab":
+        column_to_validate = "pcr_result_classification"
+    else:
+        print(f"Error: {process_type} isn't a valid option")
```
Comment (osborj) [inline cishouseholds/merge.py:272]: Get rid

**Comment by osborj** [comment]
Code coverage fails as route of 'swab' or 'antibody' is essentially indistinguishable and is not deemed to be necessary to be covered by an extra test.

**Review by howelj — APPROVED** [review]
Code meets required standards

**Pull request merged**